### ISSUE TITLE
Look into adding filters to Ample

### ISSUE THREAD

**Issue by keith-turner** [apache/accumulo#3933]
As outlined in [this comment](https://github.com/apache/accumulo/pull/3929#discussion_r1382616473) it may be useful to add filters to Ample.  These filters could configure an iterator on the scanner to do server side filtering.

A use case for this could be the GarbageCollectWriteAheadLogs class which needs to find all tablets with write ahead logs.  In #3904 the WAL GC class was moved away from using the TabletManagmentIterator to just using Ample.  This was done because the TabletMgmtIter is tightly coupled to the TabletGroupWatcher and using it elsewhere may introduce bugs (because when it is modified, only the TGW uses cases may be considered).  When GarbageCollectWriteAheadLogs was using the TabletManagmentIterator it was getting some server side filtering (although it was tricky to reason about the correctness of that filtering).

Looking at [this code](https://github.com/apache/accumulo/blob/84419e33fc601e52626df25a1032dc0a20488eac/server/gc/src/main/java/org/apache/accumulo/gc/GarbageCollectWriteAheadLogs.java#L85) maybe it could be modified to look like the following.

```java
    this.store = () -> Iterators.concat(
        context.getAmple().readTablets().forLevel(DataLevel.ROOT).filter(new HasWalsFilter())
            .fetch(LOCATION, LAST, LOGS, PREV_ROW, SUSPEND).checkConsistency().build().iterator(),
        context.getAmple().readTablets().forLevel(DataLevel.METADATA).filter(new HasWalsFilter())
            .fetch(LOCATION, LAST, LOGS, PREV_ROW, SUSPEND).checkConsistency().build().iterator(),
        context.getAmple().readTablets().forLevel(DataLevel.USER).filter(new HasWalsFilter())
            .fetch(LOCATION, LAST, LOGS, PREV_ROW, SUSPEND).checkConsistency().build().iterator());
```

In the example above `.filter(new HasWalsFilter())` add a custom filter to the Ample read that knows how to only returns tablet metadata rows from the server side that have write ahead logs.

**Comment by DomGarguilo** [comment]
Im going to assign myself as I plan to take a look into this ticket

**Comment by DomGarguilo** [comment]
Fixed via #3968
